packages/node-experimental/src/sdk/api.ts

```diff
@@ -0,0 +1,225 @@
+    : mergedBreadcrumb;
+
+  if (finalBreadcrumb === null) return;
+
+  if (client.emit) {
+    client.emit('beforeAddBreadcrumb', finalBreadcrumb, hint);
+  }
+
+  getIsolationScope().addBreadcrumb(finalBreadcrumb, maxBreadcrumbs);
+}
+
+/**
+ * Add a global event processor.
+ */
+export function addGlobalEventProcessor(eventProcessor: EventProcessor): void {
+  getGlobalScope().addEventProcessor(eventProcessor);
+}
+
+/**
+ * Add an event processor to the current isolation scope..
+ */
+export function addEventProcessor(eventProcessor: EventProcessor): void {
+  getIsolationScope().addEventProcessor(eventProcessor);
+}
+
+/** Set the user for the current isolation scope. */
+export function setUser(user: User | null): void {
+  getIsolationScope().setUser(user);
+}
+
+/** Set tags for the current isolation scope. */
+export function setTags(tags: { [key: string]: Primitive }): void {
+  getIsolationScope().setTags(tags);
+}
+
+/** Set a single tag user for the current isolation scope. */
+export function setTag(key: string, value: Primitive): void {
+  getIsolationScope().setTag(key, value);
+}
+
+/** Set extra data for the current isolation scope. */
+export function setExtra(key: string, extra: Extra): void {
+  getIsolationScope().setExtra(key, extra);
+}
+
+/** Set multiple extra data for the current isolation scope. */
+export function setExtras(extras: Extras): void {
+  getIsolationScope().setExtras(extras);
+}
+
+/** Set context data for the current isolation scope. */
+export function setContext(
+  name: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  context: { [key: string]: any } | null,
+): void {
+  getIsolationScope().setContext(name, context);
+}
+
+/** Start a session on the current isolation scope. */
+export function startSession(context?: Session): Session {
+  const client = getClient();
+  const scope = getIsolationScope();
+
+  const { release, environment = DEFAULT_ENVIRONMENT } = client.getOptions();
+
+  // Will fetch userAgent if called from browser sdk
+  const { userAgent } = GLOBAL_OBJ.navigator || {};
+
+  const session = makeSession({
+    release,
+    environment,
+    user: scope.getUser(),
+    ...(userAgent && { userAgent }),
+    ...context,
+  });
+
+  // End existing session if there's one
+  const currentSession = scope.getSession && scope.getSession();
+  if (currentSession && currentSession.status === 'ok') {
+    updateSession(currentSession, { status: 'exited' });
+  }
+  endSession();
+
+  // Afterwards we set the new session on the scope
+  scope.setSession(session);
+
+  return session;
+}
+
+/** End the session on the current isolation scope. */
+export function endSession(): void {
+  const scope = getIsolationScope();
+  const session = scope.getSession();
+  if (session) {
+    closeSession(session);
+  }
+  _sendSessionUpdate();
+
+  // the session is over; take it off of the scope
+  scope.setSession();
```

Comment:
I would inline _sendSessionUpdate into this function and unset the current session before client.captureSession is called to avoid race conditions.

Reply:
I think I'll leave this for now and refactor all the session things in a follow up, as that can generally be improved I think. But it should work somewhat the same as it did before now!